### PR TITLE
Consistent techniques

### PR DESCRIPTION
**Pull request by plazma-rush**
I've noticed some inconsistencies in what techniques are assigned to what weapons, so I thought I'd take a crack at trying to figure out some sort of logic for what weapons get what techniques, and modifying them based on that.
These rough criteria were mostly based off of what techniques the various weapons already have, with some modification. If you have any questions about my justification for some of the changes I've made, feel free to ask

Block:
-Anything sturdy enough to take multiple hits and large enough to reliably block them. Most real melee weapons larger than a knife should have at least this.

Parry:
-Same criteria as Block, but also maneuverable enough to quickly deflect blows. More effective than just Block by virtue of being able to get the weapon into place to block faster.

Shield:
-Full-sized shields.

Disarm:
-Weapons with hooks/prongs/etc capable of catching weapons.

Spinning Strike:
-Long, double-sided weapons.
-Longer lashing weapons. (how long?)

Wide Strike :
-Large, heavy weapons with lots of carry through.
-Long, bladed weapons meant for wide, sweeping blows.

Impaling Strike:
-Bladed polearms. Must have a real, sturdy spearhead/blade.

Brutal Strike:
-Large, heavy weapons with sizable bashing damage, capable of knocking people backwards through sheer weight alone.

Rapid Strike: 
-Lightweight and maneuverable weapons. Katana or lighter.
-Weapons that are light for their size.

Wrap:
-Long, heavy lashing weapons. At least enough length to wrap around a person several times.

Sweep:
-Weapons heavy enough or with enough leverage to knock out a target's legs from under them. Must also have bashing be a sizable part of their total damage. Destroying someone's legs from under them was not taken into account since such an injury would last much longer than 2 turns.

Precise Strike:
-Cut or stab weapons meant for precise, surgical strikes.

Swords in general were upgraded from Block to Parry:
- Cutlass, Wakizashi, Jian, Scimitar, Longsword, Arming Sword, and all variants upgraded from Block to Parry.
- Broadsword and Inferior variant upgraded from Block to Parry. Fake variant already had Parry.
- Monomolecular Sword upgraded from Block to Parry.
-Machete and flaming variant upgraded from Block to Parry. While not a sword, it was designed with hacking motions in mind, so it shouldn't be too unwieldy.
- Survivor Machete upgraded from Block to Parry and had Precise Strike replaced with Rapid Strike. Improved balance should be an upgrade over a normal machete, but it won't suddenly turn it into a weapon that relies on finesse like a rapier or scalpel.
- Rapier was given Precise Strike.

Large combat knives can be used to parry attacks, but not as effectively as a larger weapon. They are also light and maneuverable. Thus, they should have Block and Rapid.
- Kris was given Block and had its Brutal Strike removed. It shouldn't be able to knock someone back by itself (Silat does let you do that if you really want to, though.)
- Fake Kris was given Block.
- Kukri was given Rapid Strike.

Polearms in general had their Parry downgraded to Block. Spears are surprisingly effective up close in spear-on-sword fights, but since most of the enemies we fight attack with their body parts, the ranges even closer than that, which makes blocking difficult. Spear users should try to maintain distance and use reach attacks to keep their advantage:
- Forked Spear keeps its Parry since it's designed with defense (catching weapons/enemies) in mind.
- Sharpened rebar was given Block.
- Naginata and Glaive had Parry downgraded to Block and were given Wide Strike due to being slashing weapons.
- Survivor Naginata had its Rapid Strike removed.
- Halberd and Fake Halberd were given Sweep for their weight.
- Pitchfork had Impaling Strike replaced with Disarm. The way the four prongs are spaced out makes it difficult to go straight through someone, but they can certainly catch weapons.

Quarterstaves are shorter than polearms and are thus more maneuverable. They can also be easily used to knock out someone's legs from under them.
- Quarterstaff and Ironshod and Powered variants upgraded from Block to Parry and were also given Sweep.

Sticks and wooden weapons:
- Heavy Stick and Long Stick were given Block. Though they lack proper handles, they should at least be usable defensively.
- Bokken was given Sweep. Being a blunt weapon, knocking someone's legs out should be easy.
- Cudgel was given Parry since it's meant to be a training tool for practicing dueling moves. Though it's blunt, it's too thin for Sweep.

Batons are very lightweight and maneuverable. Thus, they should have Block or Parry depending on size, and Rapid Strike.
- Expandable Baton in its retracted form has its Block removed and was given Rapid Strike in its expanded form.
- PR-24 Baton keeps its Block technique in its compacted form due to being much larger than the Expandable Baton. It turns into Parry when extended.

Sturdy tools that can be used as weapons and makeshift weapons were given basic blocking ability:
- Hockey Stick, Makeshift Macahuitl, Digging Stick, Crowbar, Hoe, Pickaxe, Sickle, Torch, Makeshift Machete, and Shishkebab were given Block.

Larger smashing weapons are unwieldy, but have the weight to knockback and knock out legs. They also possess lots of carry through. Thus, in general they should have Block, Brutal Strike, Sweep, and Wide Strike:
- Mjolnir and Sledge Hammer were given Block and Sweep.
- Fake Battleaxe was given Block, Wide Strike, Brutal Strike, and Sweep. It's blunt and heavy, so acts more like a hammer than an axe.

Axes are similar to smashing weapons, but since they also gore their target instead of knocking them aside, they don't get Wide Strike. The Halligan Bar and Warhammer, having pointy striking points, also behave similarly:
- Wood Axe, Battle Axe, Inferior Battle Axe, Copper Axe, and Lobotomizer were given Block, Brutal Strike, and Sweep.
- Fire Axe was given Block and Sweep.
- Halligan Bar was given Block and Sweep and had its Wide Strike removed.
- Warhammer was given Block and Sweep.

Chainsaws are ridiculously unwieldy because of the way they're gripped, and also very heavy. Thus, only the Combat Chainsaw should have Block, and all chainsaws should have Sweep:
- Combat Chainsaw (off) was given Block and Sweep.
- Combat Chainsaw (on) was given Block.
- Chainsaw Lajatang (off) was given Sweep.

Misc changes:
- Bee Sting, Fungal Fighter Sting, and Wasp String had Precise Strike removed for lacking a proper grip.
- O-yoroi had its Block removed. No other suit of armor has it.
- Muffler downgraded from Parry to Block for being unwieldy.
- Grappling Hook was given Wrap and had its Rapid Strike removed. It's much too unwieldy for rapid attacks. However, the attached hook makes it possible to wrap it around somebody with an attack.
- Switchblade was given Rapid Strike. It's too small for Block, though.
- Zweihander, Inferior Zweihander, and flaming Zweihander were given Sweep by virtue of their weight.
- Fake Zweihander was given Brutal Strike and Sweep to match the normal Zweihander since it's still pretty heavy.
- Nodachi and Fake Nodachi were given Block.

Bundled mod changes:
- Fictional Weapons had its Gunknife downgraded from Parry to Block, in comparison to the Gunsword, which keeps its Parry.
- Medieval Weapons has its Ji lose its Impaling Strike. It has a perpendicular blade attached to it so cannot as easily run someone through.
- Medieval Weapons has its Dusksword, Xiphos, Khpoesh, and Dao upgraded from Block to Parry.

A few extra notes:
- Fake versions of weapons were given Parry if their normal and inferior counterparts have it as well. There are many commonly used aluminum alloys that are strong enough to take a beating. They still would make for poor weapons, though.
- Techniques that are possible to perform with a certain weapon, but not as easily were not added. For example, Sweep could be conceivably done with any of the swords, but not as easily as with, say, a warhammer. Stuff like this would probably fit better as part of a weapon-based martial art of some sort.
- The shields included in the Medieval mod follow their own logic based on the quality of the shield in question and thus were unchanged.

I also have a few points open for discussion:
- The qiang currently possesses Precise Strike, presumably due to its reputation as a martial arts weapon. However, no other similar polearm also possesses Precise Strike. Should Precise Strike be removed from the qiang, added to other similar polearms, or be moved to some sort of Chinese weapon-based martial art?
- Why do vehicle frames all have the ability to disarm people? It seems consistent and I don't know why they would have that technique in the first place, so I didn't touch them, but if there's no reason behind it I'll add their removal to this PR.
- Should the Naginata have Impaling Strike? They're slightly curved, so I'm not sure if it would be as easy to force it straight through one target and into another.

### CODE REVIEW

**Comment by Noctifer-de-Mortem** [comment]
Imo, If "block 3 = full-sized shields" then there should be a buff shields too. Most shields in Medieval Mod are already dead weight if used with pretty much any decent weapon, and buffing weapon blocking widens this gap in usefullness.

**Comment by Amariithynar** [comment]
I'm pretty sure grappling hook has/had rapid strikes because when you're holding it in your hand, it's basically just a short length of metal with spikes coming off the top, attached to a length of cordage; https://ii.cheaperthandirt.com/fcgi-bin/iipsrv.fcgi?FIF=/images/cheaperthandirt/source/20-vdt02-5538000000_1.tif&wid=575&cvt=jpeg if you're holding the cordage near the metal and swinging it in short arcs, you can get pretty fast swings out of it, though they won't be very high impact swings, either. 

Vehicle frames being able to disarm makes sense in that you're smashing them with a massive chunk of framework. You basically just bash their weapon out of their hand, and no one's really going to hold onto it when you're swinging a 20 kg steel frame around- or even a lighter, 4 kg wooden frame.

Naginata are swords on spear handles, basically (and in fact that's how many of them were made; there's a recipe in-game that uses a katana as a primary component, in reference to this). While they can impale, they can't really impale straight through, like a more optimally shaped spear can, I'd argue.

The Qiang likely possesses Precise Strike because it is relatively small, lightweight, and used for precise stabbing. https://www.youtube.com/watch?v=FAY2LYoYCAU & https://www.youtube.com/watch?v=oO-SaK7-IJ0 It's basically a short, somewhat thin spear, where the shaft is made from a hard wood if it's meant for heavy combat, or a more flexible, lighter wood called wax wood if it's meant for wushu (latter matches the Drunken Master qiang, former matches the Hand of Death qiang). There isn't really a "short spear" non-asian weapon, but if there was, it would probably be accurate to say that it gets the "Precise Strike" function as well.

Does this cover making sure weapons have the correct range of weapon-based martial arts attached to them? I hope so, since it's about melee techniques, and styles seem like viable concern for techniques; as such I'll be listing what I think should be added here. If not, let me know and I'll split it off to its own topic.

I've noticed a couple weapons that cannot be used for Silat, for example the copper spear and forked spear; copper knife; bokken (which are basically just sword-shaped bludgeons); wakazashi aren't that dissimilar to the blades used in Silat, same with cavalry sabers and cutlasses (which are little more than more formal machete) and axes, from fire axe down to hatchets (https://www.youtube.com/watch?v=Ir2tnu3ZZog). Eskrima (Arnis Kali) should be able to use all the same weapons as Silat, as they are functionally quite similar in their "use what is at hand, especially bludgeoning/generally light bladed weapons" concepts, though Eskrima definitely does support more improvised weapons. Here's Arnis spear, for example: https://www.youtube.com/watch?v=Rvq9ANCLekc and spear & walking staff: https://www.youtube.com/watch?v=1HQdPyUMpXY

Broadswords can be used with Fencing, perplexingly, yet Arming Swords, their lighter, slimmer brethren that usually have a better taper to the point, cannot (not sure if neither or both or a swap should be done here). Jian are long, thin swords that definitely can be used similar to rapiers, epees, and sabres, so should be matched to Fencing. 

Krav Maga should be usable with pistols (not shooting with it, but if you have to melee with a pistol equipped; treat it as an auto "force unarmed" perhaps, without losing the style bonus?), combat-ready knives (similar to Silat/Eskrima), and other short bludgeons than tonfa, like the batons and the cudgel. http://www.your-krav-maga-expert.com/krav-maga-weapons.html

**Comment by Amariithynar** [comment]
>Imo, If "block 3 = full-sized shields" then there should be a buff shields too. Most shields in Medieval Mod are already dead weight if used with pretty much any decent weapon, and buffing weapon blocking widens this gap in usefullness.

Shields are mostly for dealing with other weapons, not to stop grasping, flailing limbs; they can do so, but not as effectively, considering limb flexibility. Furthermore, shields *should* have High Blocking Ability, not just Medium Blocking Ability.

**Comment by Shodan14** [comment]
Regarding vehicle frame disarming, I always saw it as using something like a car door that has a lot of different sized holes in it to catch weapons (see here: [https://previews.123rf.com/images/2dmolier/2dmolier1612/2dmolier161200445/67860259-car-door-frame-on-white-background-3d-illustration.jpg](url) ).

File: data/json/items/tools.json [hunk 1]
```diff
@@ -2186,6 +2191,7 @@
     "material": "wood",
     "symbol": "/",
     "color": "brown",
+	"techniques": [ "WBLOCK_1" ],
```
Comment (DracoGriffin) [inline data/json/items/tools.json]: Inappropriate tab here, replace with four spaces for styling

File: data/json/items/tools.json [hunk 2]
```diff
@@ -4605,6 +4613,7 @@
     "symbol": "/",
     "color": "light_gray",
     "qualities": [ [ "CUT", 1 ], [ "BUTCHER", -27 ] ],
+	"techniques": [ "WBLOCK_1" ],
```
Comment (DracoGriffin) [inline data/json/items/tools.json]: Inappropriate tab here, replace with four spaces for styling

File: data/json/items/tools.json [hunk 3]
```diff
@@ -6958,6 +6972,7 @@
     "symbol": ";",
     "color": "dark_gray",
     "qualities": [ [ "CUT", 1 ], [ "CUT_FINE", 1 ], [ "BUTCHER", 14 ] ],
+	"techniques": [ "RAPID" ],
```
Comment (DracoGriffin) [inline data/json/items/tools.json]: Inappropriate tab here, replace with four spaces for styling

File: data/json/items/tools.json [hunk 4]
```diff
@@ -5401,6 +5410,7 @@
     "symbol": "/",
     "color": "dark_gray",
     "use_action": "PICKAXE",
+    "techniques" : [ "WBLOCK_1" ],
```
Comment (DracoGriffin) [inline data/json/items/tools.json]: Inappropriate space here `"techniques" :`, should be `"techniques":`

I know some jsons aren't consistent about this (most likely haven't been formatted).

File: data/json/items/tools.json [hunk 5]
```diff
@@ -6404,6 +6414,7 @@
       "failure_message": "Aw, dangit.  It fails to start!",
       "lacks_fuel_message": "This thing needs some fuel!"
     },
+    "techniques" : [ "WBLOCK_1" ],
```
Comment (DracoGriffin) [inline data/json/items/tools.json]: Inappropriate space here `"techniques" :`, should be `"techniques":`

File: data/json/items/tools.json [hunk 6]
```diff
@@ -6442,6 +6453,7 @@
         "moves": 30
       }
     ],
+    "techniques" : [ "WBLOCK_1" ],
```
Comment (DracoGriffin) [inline data/json/items/tools.json]: Inappropriate space here `"techniques" :`, should be `"techniques":`

File: data/json/items/tools.json [hunk 7]
```diff
@@ -6462,7 +6474,7 @@
     "ammo": "battery",
     "max_charges": 500,
     "charges_per_use": 100,
-    "techniques": [ "WBLOCK_1", "RAPID" ],
+    "techniques" : [ "WBLOCK_2", "RAPID", "SWEEP" ],
```
Comment (DracoGriffin) [inline data/json/items/tools.json]: Inappropriate space here `"techniques" :`, should be `"techniques":`

File: data/json/items/melee.json [hunk 8]
```diff
@@ -90,7 +93,7 @@
         "color" : "brown",
         "symbol" : "/",
         "material" : ["wood"],
-        "techniques" : ["WBLOCK_1", "RAPID", "PRECISE"],
+        "techniques" : ["WBLOCK_2", "RAPID", "PRECISE"],
```
Comment (DracoGriffin) [inline data/json/items/melee.json]: Looks like this json hasn't been formatted properly, so there are tons of format issues (which aren't your fault nor did you add them). So not sure how big a deal of these are unless someone else asks for it to be run through the [linter](http://dev.narc.ro/cataclysm/format.html).

**Comment by DracoGriffin** [comment]
Appreciate the work to go through all of this and detailing why techniques should be given to items.

Nice job!

File: data/json/items/melee.json (hunk 8 above)
Comment (plazma-rush) [reply data/json/items/melee.json]: Should I wait and make a separate PR that lints melee.json after this is merged or closed, or lint it as part of this PR?

Comment (DracoGriffin) [reply data/json/items/melee.json]: Honestly, I'd hazard on the side of linting in a separate PR unless a more senior community member thinks it's better to lint within this PR.

**Comment by Inglonias** [comment]
Look forward to messing around with this. Good job.

File: data/json/items/melee.json (hunk 8 above)
Comment (ZhilkinSerg) [reply data/json/items/melee.json]: Next PR will be better.

**Comment by Blank99** [comment]
> -Lightweight and maneuverable weapons. Katana or lighter.
Contrary to popular opinion, katana are not light. They weight about the same as your average arming sword or your average basket-hilt sword (not sure which of the two is being used as 'broadsword'.

Oddly, the game's longsword is unusually heavy. While still inside the weight range for them, its' on the extreme end.

**Comment by DracoGriffin** [comment]
>Contrary to popular opinion, katana are not light. They weight about the same as your average arming sword or your average basket-hilt sword (not sure which of the two is being used as 'broadsword'.
A scimitar might also be on the same weight range, tho it's extremely curved shape makes it's center of mass closer to the hilt, thus, easier to wield.
Oddly, the game's longsword is unusually heavy. While still inside the weight range for them, it's on the extreme end.

This sounds decent for an Issue or potentially PR if you can source the appropriate places. Quick Wiki check seems to agree [katana](https://en.wikipedia.org/wiki/Katana) and [arming sword](https://en.wikipedia.org/wiki/Knightly_sword) are similar in weights, so I think using the plazma-rush's outlines for technique attribution would work well when discussing changing the weapons you point out.

**Comment by plazma-rush** [comment]
Would the two-handed grip make a difference? The katana has about the weight of a one-handed sword, but is meant to be used with two hands. You could reposition the blade for follow up strikes with two hands, rather than just one.

**Comment by Blank99** [comment]
It would, if you know how to use the weapon right. Similarly, the longsword is not a slow weapon despite being a little heavier.

Granted, the arming sword and broadsword weren't meant to be wielded alone, they were meant to be wielded with a shield, buckler, dagger or some other defensive implement.

Oddly enough, the one style we have for katanas is one where you would use a katana one-handed, alongside a wakizashi.

**Pull request merged**